src/ProbabilisticProgramming/Loops.dfy

```diff
@@ -40,20 +40,20 @@ module Loops {
   // This definition is opaque because the details are not very useful.
   // For proofs, use the lemma `WhileUnroll`.
   // Equation (3.25), but modified to use `Monad.Diverging` instead of HOL's `arb` in case of nontermination
-  // TODO: While(condition, body)(init) would be cleaner
-  opaque ghost function While<A>(condition: A -> bool, body: A -> Monad.Hurd<A>, init: A): (f: Monad.Hurd<A>)
-    ensures forall s: Rand.Bitstream :: !condition(init) ==> f(s) == Monad.Return(init)(s)
+  opaque ghost function While<A(!new)>(condition: A -> bool, body: A -> Monad.Hurd<A>): (loop: A -> Monad.Hurd<A>)
```

Comment:
I can't quite see what has caused this change yet. How does it make our life easier if we use this curried version of the current `While`?

Reply:
I think it's easier to view a while loop as transforming a value `a: A`. If you look at the diff, it was used as a function `(init: A) => While(condition, body, init)` in many places in the file. The curried version can just be written as `While(condition, body)`.

Reply:
But I'm not super attached to that change, so if you don't like it, I can undo it.

Reply:
Ah, makes sense!